ht-12/index.js

```diff
@@ -0,0 +1,64 @@
+function carousel () {
+  const nextButtons = Array.from(document.querySelectorAll(`[data-slide="next"]`));
+  const prevButtons = Array.from(document.querySelectorAll(`[data-slide="prev"]`));
+  let directionPrev;
+  let directionNext;
+  nextButtons.forEach(button => {
+    button.addEventListener('click', () => {
+      directionDetermination(button);
+    });
+  });
+  prevButtons.forEach(button => {
+    button.addEventListener('click', () => {
+      directionDetermination(button);
+    });
+  });
+  function directionDetermination(button) {
+    const buttonTarget = button.dataset.target;
+    const buttonSlide = button.dataset.slide;
+    const carouselWrapper = document.querySelector(buttonTarget);
+    const carouselItems = Array.from(carouselWrapper.querySelectorAll('.carousel-item'));
+    if(buttonSlide === 'next') {
+      directionPrev = 'left';
+      directionNext = 'right';
+      slide(carouselItems, buttonSlide);
+    } else {
+      directionPrev = 'right';
+      directionNext = 'left';
+      slide(carouselItems, buttonSlide);
+    }
```

Comment:
refactor this code to avoid copy-paste.
or move this logic to the slide function